cvmfs/sync_mediator.cc

```diff
@@ -775,47 +774,109 @@ void SyncMediator::CreateNestedCatalog(SharedPtr<SyncItem> directory) {
 
 void SyncMediator::RemoveNestedCatalog(SharedPtr<SyncItem> directory) {
   const std::string notice =  "Nested catalog at " + directory->GetUnionPath();
-  PrintChangesetNotice(kRemoveCatalog, notice);
+  reporter_->OnRemove(notice, catalog::DirectoryEntry());
 
   if (!params_->dry_run) {
     catalog_manager_->RemoveNestedCatalog(directory->GetRelativePath());
   }
 }
 
+void SyncDiffReporter::OnInit(const history::History::Tag & /*from_tag*/,
+                              const history::History::Tag & /*to_tag*/) {}
 
-void SyncMediator::PrintChangesetNotice(const ChangesetAction  action,
-                                        const std::string     &extra) const {
-  if (!params_->print_changeset) {
-    ++changed_items_;
-    if (changed_items_ % processing_dot_interval == 0) {
-      LogCvmfs(kLogPublish, kLogStdout | kLogNoLinebreak, ".");
-    }
-    return;
+void SyncDiffReporter::OnStats(const catalog::DeltaCounters & /*delta*/) {}
+
+void SyncDiffReporter::OnAdd(const std::string &path,
+                             const catalog::DirectoryEntry & /*entry*/) {
+  changed_items_++;
+  AddImpl(path);
+}
+void SyncDiffReporter::OnRemove(const std::string &path,
+                                const catalog::DirectoryEntry & /*entry*/) {
+  changed_items_++;
+  RemoveImpl(path);
+}
+void SyncDiffReporter::OnModify(const std::string &path,
+                                const catalog::DirectoryEntry & /*entry_from*/,
+                                const catalog::DirectoryEntry & /*entry_to*/) {
+  changed_items_++;
+  ModifyImpl(path);
+}
+
+void SyncDiffReporter::CommitReport() {
+  if (changed_items_ >= processing_dot_interval_) {
```

Comment:
Now that we removed the `if` in `SyncMediator::Commit`, we need to check here if we print the changeset (no extra newline) or if we print dots (extra newline if `changed_items_ >= processing_dot_interval`).